internal/kafka/internal/handlers/validation.go

```diff
@@ -51,46 +51,109 @@ func ValidateKafkaClusterNameIsUnique(name *string, kafkaService services.KafkaS
+	supportedProviders := providerConfig.ProvidersConfig.SupportedProviders
 
-		// Set Cloud Region default if not received in the request
-		if kafkaRequest.Region == "" {
-			defaultRegion, _ := provider.GetDefaultRegion()
-			kafkaRequest.Region = defaultRegion.Name
-		}
+	defaultProvider, _ := supportedProviders.GetDefault()
+	providerName := arrays.FirstNonEmptyOrDefault(defaultProvider.Name, kafkaRequest.CloudProvider)
+	// Validation for Cloud Provider
+	provider, providerSupported := supportedProviders.GetByName(providerName)
+	if !providerSupported {
+		return "", "", errors.ProviderNotSupported("provider %s is not supported, supported providers are: %s", kafkaRequest.CloudProvider, supportedProviders)
+	}
 
-		// Validation for Cloud Region
+	// Validation for Cloud Region
+	if kafkaRequest.Region != "" { // if region is empty, default region will be chosen, so no validation is needed
 		regionSupported := provider.IsRegionSupported(kafkaRequest.Region)
 		if !regionSupported {
-			return errors.RegionNotSupported("region %s is not supported for %s, supported regions are: %s", kafkaRequest.Region, kafkaRequest.CloudProvider, provider.Regions)
+			return "", "", errors.RegionNotSupported("region %s is not supported for %s, supported regions are: %s", kafkaRequest.Region, kafkaRequest.CloudProvider, provider.Regions)
 		}
+	}
+
+	claims, err := getClaims(ctx)
+	if err != nil {
+		return "", "", err
+	}
+
+	owner := auth.GetUsernameFromClaims(claims)
+	organisationId := auth.GetOrgIdFromClaims(claims)
+
+	// Validate Region/InstanceType
+	instanceType, err := (*kafkaService).AssignInstanceType(owner, organisationId)
+	if err != nil {
+		return "", "", errors.NewWithCause(errors.ErrorGeneral, err, "error assigning instance type: %s", err.Error())
+	}
+
+	var region config.Region
+	if kafkaRequest.Region == "" {
+		region, _ = provider.GetDefaultRegion()
+	} else {
+		region, _ = provider.Regions.GetByName(kafkaRequest.Region)
+	}
+
+	if !region.IsInstanceTypeSupported(config.InstanceType(instanceType)) {
+		return "", "", errors.InstanceTypeNotSupported("instance type '%s' not supported for region '%s'", instanceType.String(), region.Name)
+	}
 
-		// Validate Region/InstanceType
-		instanceType, err := (*kafkaService).AssignInstanceType(kafkaRequest)
+	return providerName, region.Name, nil
+}
+
+// ValidateCloudProvider returns a validator that validates provided provider and region
+func ValidateCloudProvider(ctx context.Context, kafkaService *services.KafkaService, kafkaRequest *public.KafkaRequestPayload, providerConfig *config.ProviderConfig, action string) handlers.Validate {
+	return func() *errors.ServiceError {
+		_, _, err := getCloudProviderAndRegion(ctx, kafkaService, kafkaRequest, providerConfig)
+		return err
+	}
+}
+
+func getInstanceTypeAndSize(ctx context.Context, kafkaService *services.KafkaService, kafkaConfig *config.KafkaConfig, kafkaRequestPayload *public.KafkaRequestPayload) (string, string, *errors.ServiceError) {
```

Comment:
Same comment as https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/998/files#r865011118

Reply:
same as before, the instance type is used to get the size. 
Without a validation context, they can't be splitted because to validate the size you need the instance type.
We could split and have a `getInstanceType` and a `getSize(instancetype)` but since we use them always together we wouldn't have any benefit.